clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherUtils.java

```diff
@@ -261,6 +276,73 @@ void updateSubscriptionState(Map<TopicPartition, OffsetFetcherUtils.ListOffsetDa
         }
     }
 
+    OffsetResetStrategy timestampToOffsetResetStrategy(long timestamp) {
```

Comment:
Should this be static? Ditto for a few other methods moved into this class.

Reply:
Definitely, done for a few in similar situation...